eq-author/src/App/page/Logic/Routing/RoutingPage/NoRouting.test.js

```diff
@@ -1,26 +1,211 @@
+        <NoRouting onAddRouting={jest.fn()} title="Test" page={page} questionnaire={questionnaire}>
+          Ullamcorper Venenatis Fringilla
+        </NoRouting>
+      </QuestionnaireContext.Provider>
+    );
+
+    expect(queryByText(enabledTitle)).toBeTruthy();
+    expect(queryByText(enabledParagraph)).toBeTruthy();
+    expect(queryByText(disabledTitle)).toBeFalsy();
+    expect(queryByText(disabledParagraph)).toBeFalsy();
+
+    expect(screen.getByTestId("btn-add-routing")).not.toHaveStyle(
+      `pointer-events: none; 
+       opacity: 0.6;`
+    );
+  });
+
+  it("should render with button disabled", () => {
+    let props, questionnaire, page;
+
+    questionnaire = {
+      sections: [
+        {
+          id: 1,
+          title: "Section-1",
+          position: 0,
+          pages: [
+            {
+              id: "page-1",
+              pageType: "QuestionPage",
+              title: "<p>Questions 1</p>",
+              position: 0,
+              description: "",
+              answers: [
+                {
+                  id: "ans-p1-1",
+                  description: "",
+                  guidance: "",
+                  label: "num1",
+                  questionPageId: "qp-1",
+                  secondaryLabel: null,
+                },
+              ],
+            },
+            {
+              id: "page-2",
+              pageType: "QuestionPage",
+              title: "<p>Questions 2</p>",
+              position: 1,
+              description: "",
+              answers: [
+                {
+                  id: "ans-p2-1",
+                  description: "",
+                  guidance: "",
+                  label: "num2",
+                  questionPageId: "qp-2",
+
+                  secondaryLabel: null,
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    };
+
+    page = {
+      id: "page-2",
+      pageType: "QuestionPage",
+      title: "<p>Questions 2</p>",
+      position: 1,
+      description: "",
+      answers: [
+        {
+          id: "ans-p2-1",
+          description: "",
+          guidance: "",
+          label: "num2",
+          questionPageId: "qp-2",
+
+          secondaryLabel: null,
+        },
+      ],
+    };
+
+    props = {
+      questionnaire,
+      page,
+    };
+
+    const { queryByText } = render(
+      <QuestionnaireContext.Provider value={{ props }}>
+        <NoRouting onAddRouting={jest.fn()} title="Test" otherProps={props}>
+          Ullamcorper Venenatis Fringilla
+        </NoRouting>
+      </QuestionnaireContext.Provider>
+    );
+
+    expect(queryByText(disabledTitle)).toBeTruthy();
+    expect(queryByText(disabledParagraph)).toBeTruthy();
+    expect(queryByText(enabledTitle)).toBeFalsy();
+    expect(queryByText(enabledParagraph)).toBeFalsy();
+
+    expect(screen.getByTestId("btn-add-routing")).toHaveStyle(
+      `pointer-events: none; 
+       opacity: 0.6;`
```

Comment:
The same as above.

> Does this check to see if the button isn't disabled? If so, you can use this: .toHaveAttribute('disabled') and it'll live longer. If we ever change the button styles, which might happen if their contrast becomes an issue for accessibility, then this assertion would fail if the opacity changes.

Reply:
The same as above.

> Josh:
Added disabled={isLastPage} into AddRoutingButton which prevents keyboard focus too

> Matt:
Awesome. Cheers Josh. In that case, could you update this test to assert on that instead please?